### PR TITLE
fix: portfolio dispatching (#820)

### LINKED ISSUE #820: [Alpha] Dashboard charts and assets not loading when switching wallets to the first/previous one connected
# Brief description (high level)
When connecting multiple wallets to app.shapeshift.com and switch wallet to the first/previous one connected, the Dashboard and assets does not load

*Happens with all wallets regardless of which is connected first
**Mitigation step is to hard refresh the page

#  Steps to recreate:
1. Go to app.shapeshift.com
2. Connect wallet 1 (e.g. Native - you can create one if you need to)
3. Connect wallet 2 (e.g. metamask ) by going to Menu then Switch Wallet Provider
4. Switch back to first wallet connected which is wallet 1 (e.g. Native)
5. Witness that dashboard and assets are not loading
6. Switch back to wallet 2, notice that dashboard and assets are not loading as well

#  Screenshots/vids
https://user-images.githubusercontent.com/83658503/150038586-a8f0496a-d5c5-4874-a618-ee39dee75df1.mov

#  Acceptance Criteria
- Dashboard and assets should load when switching between multiple wallets on alpha 

### **• Bounty Hunter:**
- Join our discord [here](https://discord.gg/shapeshift). Once you join you can see more about this issue [here](https://discord.com/channels/554694662431178782/873244924026449940/925799058537336853)
- Include an expected timeline for you to complete work in the work plan when you apply for this bounty!
- Please refer to [this link](https://shapeshiftdao.gitbook.io/getting-started/how-to-contribute/bounties#resources-for-new-contributors-and-bounty-hunters) for some basic info
-  Link to Discord triage once you join above - 
https://discord.com/channels/554694662431178782/873244924026449940/933148091064909826

### PR DESCRIPTION
**Pull request by amitojsingh366**
## Description

Please describe your changes

## Notice

Before submitting a pull request, please make sure you have answered the following:

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?
- [x] Do all new and existing tests pass? Does the linter pass?

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes #820 

## Testing

Please outline all testing steps

1. Pull branch locally and run `yarn dev` to start the server
2. Go to your dashboard and switch between your wallets 
3. Notice how the charts and wallet assets are now loading up

## Screenshots (if applicable)

https://user-images.githubusercontent.com/35400192/151299673-3cb9cb51-ecca-45a0-b015-0962b19c7725.mov

## Additional Info

I was asked to provide an estimate on how much time this took, and this issue took way longer than expected and the solution was a rather simple one. My rough estimate for this is around 4-5 hours

### CODE REVIEW

File: src/context/PortfolioProvider/PortfolioContext.tsx [hunk 1]
```diff
@@ -32,7 +32,7 @@ export const PortfolioProvider = ({ children }: { children: React.ReactNode }) =
     dispatch(portfolio.actions.clear())
     // fetch each account
     accountSpecifiers.forEach(accountSpecifier =>
-      dispatch(portfolioApi.endpoints.getAccount.initiate(accountSpecifier))
+      dispatch(portfolioApi.endpoints.getAccount.initiate(accountSpecifier, { forceRefetch: true }))
```
Comment (0xdef1cafe) [inline src/context/PortfolioProvider/PortfolioContext.tsx:35]: nice catch - i didn't anticipate that we'd be switching wallets quickly and this wouldn't refetch, and would never hit the `onCacheEntryAdded`

File: src/state/slices/portfolioSlice/portfolioSlice.ts [hunk 2]
```diff
@@ -182,20 +182,22 @@ export const portfolioApi = createApi({
...
           return { error }
         }
-      },
-      onCacheEntryAdded: async (_args, { dispatch, cacheDataLoaded, getCacheEntry }) => {
-        await cacheDataLoaded
-        const port = getCacheEntry().data
-        if (!port) return
-        dispatch(portfolio.actions.upsertPortfolio(port))
       }
+      // NOTE (amitojsingh366): Not sure if we need this now since I moved the dispatch in the query itself, this fixed #820
```
Comment (0xdef1cafe) [inline src/state/slices/portfolioSlice/portfolioSlice.ts:194]: cool, i understand why this is the fix, can you please delete this commented code?

**Comment by cjthompson** [comment]
I agree with @0xdef1cafe - Please add a _why_ comment to the change and remove the commented code. I'm happy to approve after that small change.

File: src/context/PortfolioProvider/PortfolioContext.tsx (hunk 1 above)
Comment (amitojsingh366) [reply src/context/PortfolioProvider/PortfolioContext.tsx:35]: yeah sure, I will leave a comment here

File: src/state/slices/portfolioSlice/portfolioSlice.ts (hunk 2 above)
Comment (amitojsingh366) [reply src/state/slices/portfolioSlice/portfolioSlice.ts:194]: Sure I'll remove the commented code, do you also want me to remove the note?

File: src/context/PortfolioProvider/PortfolioContext.tsx (hunk 1 above)
Comment (amitojsingh366) [reply src/context/PortfolioProvider/PortfolioContext.tsx:37]: done

**Review by theoboldfrazier — COMMENTED** [review]
Looks good.

**Review by 0xdef1cafe — APPROVED** [review]

**Review by mrnerdhair — APPROVED** [review]

**Review by cjthompson — APPROVED** [review]

**Pull request merged**